sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/LiveMetrics/Manager.Metrics.cs

```diff
@@ -45,9 +46,10 @@ public MonitoringDataPoint GetDataPoint()
             string projectionError = string.Empty;
             Dictionary<string, AccumulatedValues> metricAccumulators = CreateMetricAccumulators(_collectionConfiguration);
             LiveMetricsBuffer liveMetricsBuffer = new();
-            DocumentBuffer filledBuffer = _documentBuffer.FlipDocumentBuffers();
             IEnumerable<DocumentStream> documentStreams = _collectionConfiguration.DocumentStreams;
-            foreach (var item in filledBuffer.ReadAllAndClear())
+
+            ConcurrentQueue<DocumentIngress> filledBuffer = _documentBuffer.FlipDocumentBuffers();
```

Comment:
Previously we had a limit of 20. Now we don't have a limit - given LiveMetrics is before sampler, is there a risk of taking up too much memory/overflow etc?
I think its best to have a cap always.

(Sorry, I dont have full context of this issue being fixed)

Reply:
Have to get into implementation details to fully explain :)

For LiveMetrics, we send telemetry once per second. So at most there's only two buffers held at a time (one each for Writing new telemetry and Reading to transmit). Although the buffer had a max capacity, the capacity was only evaluated after allocating space for a new Document. So the current implementation wasn't meeting our goals of saving memory.

We discussed alternatives to fix this. 
Any max capacity would cause Documents to be dropped. Our current implementation calculates metrics from the Documents added to the buffer. We would need a larger refactor to calculate metrics before any documents are dropped. Additionally , if we set a capacity I think it should accommodate the 90th percentile of customer telemetry volume with some 

After discussing we decided on this quick fix, which is removing the max capacity. 
I let my sample app run for an hour with this change and didn't see unbounded memory.

Reply:
Not sure why there is a limit of 20. I didn't find an equivalent in the old SDK regarding the limit.